todomvc.js

```diff
@@ -1,102 +1,91 @@
 import "./styles.css";
 
-import React from "react";
 import ReactDOM from "react-dom";
-import route from "can-route";
-import DefineMap from "can-define/map/";
-import Component from "react-view-model/component";
+import React from "react";
+import Todo from "../models/todo";
 import Create from "./components/create";
 import List from "./components/list";
-import Todo from "./models/todo";
+import { connect, ObserveObject } from "ylem";
+import { getAsync } from 'ylem/property-decorators';
+import route from "can-route";
 
-export const ViewModel = DefineMap.extend('AppVM', {
-	filter: "string",
-	get todosPromise() {
-		if(!this.filter) {
+class Store extends ObserveObject {
+    @getAsync
+	get todosList() {
+		if(!route.data.filter) {
 			return Todo.getList({});
 		}
 		else {
-			return Todo.getList({ complete: this.filter === "complete" });
+			return Todo.getList({ filter: {complete: route.data.filter === "complete" }});
 		}
-	},
-	todosList: {
-		get: function(lastSetValue, resolve) {
-			this.todosPromise.then(resolve);
-		}
-	},
+	}
 	get allChecked() {
 		return this.todosList && this.todosList.allComplete;
-	},
+	}
 	set allChecked(newVal) {
 		this.todosList && this.todosList.updateCompleteTo(newVal);
-	}
-});
+    }
```

Comment:
You should run the included eslint, or better yet: grab the latest eslint file from the ylem repo so the examples match and run that one. I would also recommend integrating eslint into the test script (even if there aren't any real tests) so travis catches it.